examples/metalk8s-solution-example/operator/pkg/controller/util/baseserver.go

```diff
@@ -109,9 +110,12 @@ func BuildLabelSelector(labels map[string]string) metav1.LabelSelector {
 // buildImageName builds a complete image name based on the version provided
 // for the `base-server` component, which is the only one deployed for now
 func buildImageName(version string) string {
-	prefix, found := os.LookupEnv("REGISTRY_PREFIX")
-	if !found {
-		prefix = "docker.io/metalk8s"
+	var prefix string = "docker.io/metalk8s"
+
+	for _, registry := range config.OperatorConfig.Repositories {
+		if registry.Version == version {
+			prefix = registry.Repository
+		}
```

Comment:
Not entirely sure about this logic. Consider a complex Operator which not only deploys custom images (which will have the same version as the Operator, probably), but also some open-sourced components (e.g. a well-known database or queue system). The version of said images would not be the same as the `repository.Version`, yet would live in one specific repository.

Mapping between those would need to live somewhere, and I don't what would be the most flexible approach here. Thoughts @benoit-a @RachedBenMustapha?

Reply:
Yeah right, didn't though of other Solutions here.
Will try to find a better format for the repository mapping.

Reply:
@gdemonet for Zenko Operator, since we have all possible cases you mentioned (including being used outside of the solution framework), we use a combination of an image prefix (not yet populated, always empty, but used in all generated workflows) and explicit repos/versions in a [ZenkoVersoin CR](https://github.com/scality/zenko-operator/blob/master/doc/examples/zenkoversion-1.1-dev.yaml) that is going to be generated in the release process and shipped inside a version's solution ISO.